mods/dye/init.lua

```diff
@@ -20,13 +20,104 @@ dye.dyes = {
 	pink		= { name = "Pink", html = "#FF7FFF" },
 }
 
--- Define items
+----
+-- Helper functions.
+----
+
+--[[
+	Convert an html color in the format of #RRGGBB into red, green, and blue values on the range of [0..255].
+]]
+function hex2rgb(hex)
+    hex = hex:gsub("#","")
+    return tonumber("0x"..hex:sub(1,2)), tonumber("0x"..hex:sub(3,4)), tonumber("0x"..hex:sub(5,6))
+end
+
+function rgb2hex(r, g, b)
+	return string.format("#%02X%02X%02X", r, g, b)
+end
+
+
+--[[
+	Convert red, green, and blue values from the range of [0..255]
+	into hue, saturation, and value/lightness values on the range of...?
+]]
+function rgbToHsv(red, green, blue)
+	local r = red / 255
+	local g = green / 255
+	local b = blue / 255
+	local max = math.max(r, g, b)
+	local min = math.min(r, g, b)
+	local h, s, v = max, max, max
+
+	local d = max - min;
+	if max == 0 then
+		s = 0
+	else
+		s = d / max
+	end
+
+	if(max == min) then
+		h = 0; -- achromatic
+	else
+		if   max == r then
+			if g < b then
+				h = (g - b) / d + 6
+			else
+				h = (g - b) / d + 0
+			end
+		elseif max == g then
+			h = (b - r) / d + 2
+		elseif max == b then
+			h = (r - g) / d + 4
+		end
+		h = h / 6;
+	end
+	return h, s, v
+end
+
+--[[
+	Convert the hue, saturation, and value/lightness values output from rgbToHsv
+	back into red, green, and blue values on the range of [0..255].
+]]
+function hsvToRgb(h, s, v)
+	local i = math.floor(h * 6);
+	local f = h * 6 - i;
+	local p = v * (1 - s);
+	local q = v * (1 - f * s);
+	local t = v * (1 - (1 - f) * s);
+
+	local r, g, b;
+	local switch = i % 6
```

Comment:
you could probably use a table with functions to eliminate 6 branches in exchange for a table lookup and a function call.